ansible/roles/controller/templates/chainlet/chainlet-deploy.yml.tmpl.j2

```diff
@@ -44,52 +45,115 @@ spec:
         prometheus.io/port: "26660"
-        saga.xyz/stack-version: "{{ .ChainletStackVersion }}"
-        saga.xyz/stack-name: "{{ .ChainletStackName }}"
+        saga.xyz/chain-version: "{{ .ChainVersion }}"
+        saga.xyz/chain-source: "{{ .ChainSource }}"
     spec:
       restartPolicy: Always
+{{ if and .ChainGenesisImage .ChainCCVConsumer }}
+      initContainers:
+        - name: "{{ .Name }}-genesis"
+          image: "{{ .ChainGenesisImage }}"
+          imagePullPolicy: IfNotPresent
+          command: ["/root/genesis.sh"]
+          env:
+          - name: CHAIN_ID
+            value: "{{ .ChainId }}"
+          - name: MAINTAINERS
+            value: "{{ .ChainMaintainers }}"
+          - name: DENOM
+            value: "{{ .CoinDenom }}"
+          - name: GENESIS_CCV
+            value: "{{ .ChainGenesisCcv }}"
+          - name: GENESIS_TIME
+            value: "{{ .ChainGenesisTime }}"
+{% if voting_period is defined %}
+          - name: VOTING_PERIOD
+            value: "%%% voting_period %%%"
+{% endif %}
+{{ range $key, $value := .ChainEnvVars }}
+          - name: {{ $key | ToUpper }}
+            value: "{{ $value }}"
+{{ end }}
+{{ range $key, $value := .ChainParams }}
+          - name: PARAM_{{ $key | ToUpper }}
+            value: "{{ $value }}"
+{{ end }}
+          volumeMounts:
+            - name: "saga-{{ .Name }}-persistent-storage"
+              mountPath: "/root/.sagaosd"
+              subPath: sagaosd
+{{ end }}
       containers:
-        - image: "{{ .ChainletImage }}"
-          name: chainlet
+        - image: "{{ .ChainImage }}"
+          name: "{{ .Name }}"
           imagePullPolicy: IfNotPresent
           env:
-          - name: CHAINID
+          - name: UID
+            valueFrom:
+              fieldRef:
+                apiVersion: v1
+                fieldPath: metadata.uid
+          - name: CHAIN_ID
             value: "{{ .ChainId }}"
           - name: LOGLEVEL
             value: "info"
           - name: KEYNAME
             value: "chainlet-operator-key"
           - name: EXTERNAL_ADDRESS
-            value: "{{ .ChainletExternalAddress }}"
+            value: "{{ .ChainExternalAddress }}"
           - name: MAINTAINERS
-            value: "{{ .ChainletMaintainers }}"
+            value: "{{ .ChainMaintainers }}"
+          - name: DENOM
+            value: "{{ .CoinDenom }}"
+          - name: PEERS
+            value: "{{ .ChainPeers }}"
+          - name: NODE_KEY
+            value: "{{ .NodeKey }}"
+          - name: VALIDATOR_KEY
+            value: "{{ .ValidatorKey }}"
+          - name: OPTS
+            value: "--pruning %%% pruning_strategy | default('default') %%% --json-rpc.gas-cap 50000004 --api.enable=false --grpc.enable=true"
+          - name: CHAINID
+            value: "{{ .ChainId }}"
           - name: VALIDATOR_ADDRESSES_COSMOS
-            value: "{{ .ChainletValidators }}"
+            value: "%%% validator_addresses %%%"
```

Comment:
Can we determine this at deploy time now? I don't think so.

Reply:
I think the decision was to hardcode it for the CCV chains, while still using the field `genesisValidators` for the non-CCV.